paasta_tools/cli/cmds/spark_run.py

```diff
@@ -894,6 +954,16 @@ def paasta_spark_run(args):
     user_spark_opts = _parse_user_spark_args(
         args.spark_args, pod_template_path, args.enable_compact_bin_packing
     )
+
+    # Get driver-memory and cores
+    sub_cmds = args.cmd.split(" ")  # spark.driver.memory=10g
+    for cmd in sub_cmds:
+        if cmd.startswith("spark.driver.memory") or cmd.startswith(
+            "spark.driver.cores"
+        ):
+            key, value = cmd.split("=")
+            user_spark_opts[key] = value
+
```

Comment:
Is this config not always part of `--spark args '...'` in the cmd? If it is, could we not grab this out of the spark_args that we parsed just above this? It seems like [_parse_user_spark_args](https://github.com/Yelp/paasta/blob/012a17ea344e6c932a05607af105a03fdf2ed71c/paasta_tools/cli/cmds/spark_run.py#L545-L555) would just parse anything out of there and throw it into `user_spark_opts`?

Reply:
This was added to cover all edge cases.
If users doesn't provide driver.cores as part of `--spark-args` but provide as part of `spark-submit` command, it would fail.
Example: `spark-submit --conf spark.driver.cores=12g` would set container memory as `2g`. 
I caught this in detailed manual runs, details in the ticket comment.

Happy to revisit, if I still missed something :)

Reply:
Ah, I didn't realize people could specify this in the spark-submit command instead, thanks for the explanation.

Reply:
Could you add a comment indicating this for future us?

Reply:
Added.